File: src/ReadStorm.Application/Abstractions/IReaderAutoDownloadPlanner.cs
using ReadStorm.Domain.Models;

namespace ReadStorm.Application.Abstractions;

/// <summary>
/// 根据阅读位置和章节状态生成自动下载计划。
/// </summary>
public interface IReaderAutoDownloadPlanner
{
    Task<ReaderAutoDownloadPlan> BuildPlanAsync(
        string bookId,
        int anchorChapterIndex,
        int batchSize,
        int lowWatermark,
        CancellationToken cancellationToken = default);
}

File: pureopus/app/src/main/java/com/readstorm/app/application/abstractions/IReaderAutoDownloadPlanner.kt
package com.readstorm.app.application.abstractions

import com.readstorm.app.domain.models.ReaderAutoDownloadPlan

interface IReaderAutoDownloadPlanner {
    suspend fun buildPlan(
        bookId: String,
        anchorChapterIndex: Int,
        batchSize: Int,
        lowWatermark: Int
    ): ReaderAutoDownloadPlan
}
